config/vlan.py

```diff
@@ -189,97 +231,159 @@ def vlan_member():
+        ctx.fail("{} cannot have more than one untagged Vlan.".format(port))
     
     config_db = ValidatedConfigDBConnector(db.cfgdb)
+
     if ADHOC_VALIDATION:
-        if not clicommon.is_vlanid_in_range(vid):
-            ctx.fail("Invalid VLAN ID {} (1-4094)".format(vid))
-
-        if clicommon.check_if_vlanid_exist(db.cfgdb, vlan) == False:
-            ctx.fail("{} does not exist".format(vlan))
-
-        if clicommon.get_interface_naming_mode() == "alias": # TODO: MISSING CONSTRAINT IN YANG MODEL
-            alias = port
-            iface_alias_converter = clicommon.InterfaceAliasConverter(db)
-            port = iface_alias_converter.alias_to_name(alias)
-            if port is None:
-                ctx.fail("cannot find port name for alias {}".format(alias))
-
-        if clicommon.is_port_mirror_dst_port(db.cfgdb, port): # TODO: MISSING CONSTRAINT IN YANG MODEL
-            ctx.fail("{} is configured as mirror destination port".format(port))
-
-        if clicommon.is_port_vlan_member(db.cfgdb, port, vlan): # TODO: MISSING CONSTRAINT IN YANG MODEL
-            ctx.fail("{} is already a member of {}".format(port, vlan))
-
-        if clicommon.is_valid_port(db.cfgdb, port):
-            is_port = True
-        elif clicommon.is_valid_portchannel(db.cfgdb, port):
-            is_port = False
-        else:
-            ctx.fail("{} does not exist".format(port))
-
-        if (is_port and clicommon.is_port_router_interface(db.cfgdb, port)) or \
-           (not is_port and clicommon.is_pc_router_interface(db.cfgdb, port)): # TODO: MISSING CONSTRAINT IN YANG MODEL
-            ctx.fail("{} is a router interface!".format(port))
+        for vid in vid_list:
+
+            vlan = 'Vlan{}'.format(vid)
+
+            # default vlan checker
+            if vid == 1:
+                ctx.fail("{} is default VLAN".format(vlan))
+
+            log.log_info("'vlan member add {} {}' executing...".format(vid, port))
+
+            if not clicommon.is_vlanid_in_range(vid):
+                ctx.fail("Invalid VLAN ID {} (2-4094)".format(vid))
+
+            if clicommon.check_if_vlanid_exist(db.cfgdb, vlan) == False:
+                log.log_info("{} does not exist".format(vlan))
+                ctx.fail("{} does not exist".format(vlan))
+
+            if clicommon.get_interface_naming_mode() == "alias":  # TODO: MISSING CONSTRAINT IN YANG MODEL
+                alias = port
+                iface_alias_converter = clicommon.InterfaceAliasConverter(db)
+                port = iface_alias_converter.alias_to_name(alias)
+                if port is None:
+                    ctx.fail("cannot find port name for alias {}".format(alias))
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
+            if clicommon.is_port_mirror_dst_port(db.cfgdb, port):
+                ctx.fail("{} is configured as mirror destination port".format(port))
+
+            # TODO: MISSING CONSTRAINT IN YANG MODEL
```

Comment:
same as above

Reply:
Please note that these are exisiting YANG Constraints and our proposed changes/YANG Model has not modified any of these constraints.